mk/rabbitmq-tools.mk

```diff
@@ -390,3 +390,32 @@ clean-ct-logs-archive::
 	$(gen_verbose) rm -f $(PROJECT)-ct-logs-*.tar.xz
 
 clean:: clean-ct-logs-archive
+
+# --------------------------------------------------------------------
+# Generate a file listing RabbitMQ component dependencies and their
+# Git commit hash.
+# --------------------------------------------------------------------
+
+rmq-deps.mk: $(PROJECT)-rmq-deps.mk
+	@:
+
+closing_paren := )
+
+define rmq_deps_mk_line
+dep_$(1) := git $(dir $(RABBITMQ_UPSTREAM_FETCH_URL))$(call rmq_cmp_repo_name,$(1)).git $$(git -C "$(2)" rev-parse HEAD)
+endef
+
+$(PROJECT)-rmq-deps.mk: $(ERLANG_MK_RECURSIVE_DEPS_LIST)
+	$(gen_verbose) :> $@
+	$(verbose) cat $(ERLANG_MK_RECURSIVE_DEPS_LIST) | \
+	while read -r dir; do \
+	  component=$$(basename "$$dir"); \
+	  case "$$component" in \
+	  $(foreach component,$(RABBITMQ_COMPONENTS),$(component)$(closing_paren) echo "$(call rmq_deps_mk_line,$(component),$$dir)" ;;) \
+	  esac; \
+	done > $@
```

Comment:
Did you mean `>>` here ?

Reply:
`>` is fine, this is the output of the entire `while` loop. I can probably get rid of the initial `:> $@` which is a left-over from previous trials of the implementation.